internal/deps/deps.go

```diff
@@ -0,0 +1,166 @@
+	if postgresDockerImage != "" {
+		config.postgresDockerImage = postgresDockerImage
+	} else {
+		config.postgresDockerImage = defaultPostgresDockerImage
+	}
+	return config
+}
+
+func (config *DepsConfig) WithPostgresPort(postgresPort string) *DepsConfig {
+	if postgresPort != "" {
+		config.postgresPort = postgresPort
+	} else {
+		config.postgresPort = defaultPostgresPort
+	}
+
+	return config
+}
+
+func (config *DepsConfig) WithPostgresPassword(postgresPassword string) *DepsConfig {
+	if postgresPassword != "" {
+		config.postgresPassword = postgresPassword
+	} else {
+		config.postgresPassword = defaultPostgresPassword
+	}
+	return config
+}
+
+func (config *DepsConfig) WithDevenetDockerImage(devnetDockerImage string) *DepsConfig {
+	if devnetDockerImage != "" {
+		config.devnetDockerImage = devnetDockerImage
+	} else {
+		config.devnetDockerImage = defaultDevnetDockerImage
+	}
+
+	return config
+}
+
+func (config *DepsConfig) WithDevenetPort(devnetPort string) *DepsConfig {
+	if devnetPort != "" {
+		config.devnetPort = devnetPort
+	} else {
+		config.devnetPort = defaultDevnetPort
+	}
+
+	return config
+}
+
+type DepsContainers struct {
+	postgres testcontainers.Container
+	devnet   testcontainers.Container
+}
+
+func (depsContainers *DepsContainers) toContainerArray() []testcontainers.Container {
+	return []testcontainers.Container{depsContainers.postgres, depsContainers.devnet}
+}
+
+func Run(ctx context.Context, depsConfig DepsConfig) (*DepsContainers, error) {
+
+	// wait strategy copied from testcontainers docs
+	postgresWaitStrategy := wait.ForLog("database system is ready to accept connections").
+		WithOccurrence(numPostgresCheckReadyAttempts).
+		WithPollInterval(fiveSeconds)
+
+	postgresReq := testcontainers.ContainerRequest{
+		Image: depsConfig.postgresDockerImage,
+		ExposedPorts: []string{strings.Join([]string{
+			depsConfig.postgresPort, ":5432/tcp"}, "")},
+		WaitingFor: postgresWaitStrategy,
+		Name:       "rollups-node-dep-postgres",
+		Env: map[string]string{
+			"POSTGRES_PASSWORD": depsConfig.postgresPassword,
+		},
+	}
+
+	postgres, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: postgresReq,
+		Started:          true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	devNetReq := testcontainers.ContainerRequest{
+		Image:        depsConfig.devnetDockerImage,
+		ExposedPorts: []string{strings.Join([]string{depsConfig.devnetPort, ":8545/tcp"}, "")},
+		WaitingFor:   wait.ForExec([]string{"eth_isready"}),
+		Name:         "rollups-node-dep-devnet",
+		Env: map[string]string{
+			"ANVIL_IP_ADDR": "0.0.0.0",
+		},
+	}
+
+	devnet, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: devNetReq,
+		Started:          true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &DepsContainers{postgres, devnet}, nil
+}
+
+func Terminate(ctx context.Context, depsContainers *DepsContainers) []error {
```

Comment:
```suggestion
        <-ctx.Done()
```

Instead of having a Terminate function, you can wait for the context to be canceled in the Run function. See more details about contexts at https://www.digitalocean.com/community/tutorials/how-to-use-contexts-in-go and  https://pkg.go.dev/context.

Reply:
I tried  only relying on "context canceled/Done" to kill the containers. But the containers kept running being killed after a while, maybe by "test contairners ryuk". So now I have use a function to upon "ctx.Done" try to terminate the container using the testcontainers API. But had the same outcome. Not sure it is ok as it is now

Reply:
I will take a look later today.